generators/app/templates/src/app/components/CustomButton/index.js

```diff
@@ -7,43 +7,31 @@ import CustomText from '../CustomText';
 
 import styles from './styles';
 
-class CustomButton extends PureComponent {
-  static VARIANTS = ['borderless', 'radial', 'black', 'green', 'white', 'gray'];
-
-  customStyles = () => getCustomStyles(CustomButton.VARIANTS, this.props, styles);
-
-  customTextStyles = () => getCustomStyles(CustomButton.VARIANTS, this.props, styles, 'Content');
-
-  render() {
-    const {
-      onPress,
-      style,
-      activeOpacity,
-      icon,
-      title,
-      iconStyle,
-      disabled,
-      textStyle,
-      ...textProps
-    } = this.props;
-    // TODO: Set android ripple while pressing a button
-    return (
-      <TouchableOpacity
-        onPress={onPress}
-        style={[styles.container, this.customStyles(), style]}
-        activeOpacity={activeOpacity}
-        disabled={disabled}
-      >
-        {icon && <Image source={icon} resizeMode="contain" style={[styles.icon, iconStyle]} />}
-        {title && (
-          <CustomText {...textProps} style={[this.customTextStyles(), textStyle]}>
-            {title}
-          </CustomText>
-        )}
-      </TouchableOpacity>
-    );
-  }
-}
+const VARIANTS = ['borderless', 'radial', 'black', 'green', 'white', 'gray'];
+
+const CustomButton = props => {
+  const customStyles = useCallback(() => getCustomStyles(VARIANTS, props, styles), [props]);
```

Comment:
This might be debatable, but it might be better to avoid useCallback here, mainly because this memoization will be recalculated each time the props change, and since this component has no state this is the only reason this component will change. So basically the memoization will be performed in each render, which is the same as not having it.

But again, this is debatable. We'd have to measure performance to see if memoizing here is more expensive than not doing it, and even measuing it may not throw many results because this function is very cheap to execute. So I just wanted to leave this thought here, you may leave it like this or remove useCallback as you please